src/adapter/src/coord/introspection.rs

```diff
@@ -144,6 +144,11 @@ pub fn auto_run_on_introspection<'a, 's, 'p>(
         return TargetCluster::Active;
     }
 
+    // Auto-routing mid transaction can cause an abort due to an invalid timedomain.
```

Comment:
Does one of the added tests cover this?

Reply:
Just added one. I meant to comment that some tests were failing without this because a bunch of pg_catalog queries get auto-routed and then fail mid-transaction.